engine/src/ast/field_expr.rs

```diff
@@ -245,27 +236,35 @@ impl<'s> Expr<'s> for FieldExpr<'s> {
             };
         }
 
+        macro_rules! compile {
```

Comment:
Hmm, looking at the usage, it appears this could be a regular generic function? It doesn't seem to use anything from outside the context or do much syntactically.

Reply:
Done. I introduced the `compile_with` function (on the model of `lex` and `lex_with`) for the `LhsFieldExpr` type.